app/2.0/docs/devguide/dom-template.md

```diff
@@ -60,106 +59,183 @@ except for testing. For caveats about defining elements in the main document,
-      }
-      return memoizedTemplate;
-    }
+class BaseClass extends Polymer.Element {
+  static get template() {
+    return html`This content has been inherited from BaseClass.`;
   }
+}
+```
 
-})();
+Child class definition { .caption }
+```js
+class ChildClass extends Polymer.Element {
+
+}
 ```
 
-The following example shows how an element could wrap a superclass template with its own
-template.
+[See a working example in Plunker](http://plnkr.co/edit/vS99al?p=preview).
 
-```html
-<dom-module id="my-ext">
-    <template>
-      <h2>Extended template</h2>
-      <!-- superclass template will go here -->
-      <div id="footer">
-        Extended template footer.
-      </div>
-    </template>
-  <script>
-    (function() {
-      let memoizedTemplate;
+#### Override a template definition {#overridebase}
+
+Base class definition { .caption }
 
-      class MyExtension extends MySuperClass {
+```js
+const html = Polymer.html;
 
-        static get is() { return 'my-ext'}
-        static get template() {
-          if (!memoizedTemplate) {
-            // Retrieve this element's dom-module template
-            memoizedTemplate = Polymer.DomModule.import(this.is, 'template');
+class BaseClass extends Polymer.Element {
+  static get template() {
+    return html`This content has been inherited from BaseClass.`;
+  }
+}
+```
 
-            // Clone the contents of the superclass template
-            let superTemplateContents = document.importNode(MySuperClass.template.content, true);
+Child class definition { .caption }
+```js
+const html = Polymer.html;
 
-            // Insert the superclass contents
-            let footer = memoizedTemplate.content.querySelector('#footer');
-            memoizedTemplate.content.insertBefore(superTemplateContents, footer);
-          }
-          return memoizedTemplate;
-        }
-      }
-      customElements.define(MyExtension.is, MyExtension);
-    })();
-  </script>
-</dom-module>
+class ChildClass extends BaseClass {
+  static get template() {
+    return html`
+      Base class template has been overridden. Hello from ChildClass!
+      `;
+  }
+}
+```
+
+[See a working example in Plunker](http://plnkr.co/edit/VCwhuA?p=preview).
+
+#### Extend a base class template {#extendbase}
+
+Base class definition { .caption }
+
+```js
+const html = Polymer.html;
+
+class BaseClass extends Polymer.Element {
+  static get template() {
+    return html`
+      This content has been inherited from BaseClass.
+      `;
+  }
+}
+```
+
+Child class definition { .caption }
+```js
+const html = Polymer.html;
+
+class ChildClass extends BaseClass {
+  static get template() {
+    return html`
+    <p>Hello, this content is in ChildClass's template.</p>
+    ${super.template}
+    <p>Hello again from ChildClass.</p>
+    `;
+  }
+}
 ```
 
+[See a working example in Plunker](http://plnkr.co/edit/f5utiX?p=preview).
+
+#### Insert content from a child class into placeholders supplied by a base class {#insertcontent}
```

Comment:
I'm not sure this is the correct title for this use case... It seems like the primary use case here is the _superclass_ use case: that is, "provide overrideable sub-templates." Or something.  Replaceable sub-templates? Allow easy template extension with modular templates?

@justinfagnani @kevinpschaaf @graynorton Anyone want to name this pattern? :D 

Just seems to me that the important part is, a superclass providing a focused way to extend its template with content. They could be placeholders, or not—for example, they could be a valid part of the superclass template that a subclass may or may not replace.

Reply:
Also: We may want to leave in one or both of the old-school template extension examples, since they're still valid and ```html`${foo}` ``` will only work in 2.4 and later.

Reply:
* "Template extension points"
* "Overridable sub-templates"

I'd probably go with the first...?